linode/manager
# feat: [UIE-8090] - DBaaS Settings and Landing Page

## Description 📝
Adding Suspend and Resume functionality to Landing Page and database details Settings Views for GA

## Changes  🔄
- Suspend Cluster button can now appear in the DatabaseSettings that displays  and Suspend Cluster dialog
- Suspend and Resume actions added to Database Landing table
- In the Database Landing table, clusters with suspending, suspended, and resuming status will not have detail links

## Target release date 🗓️
11/12/2024

## Preview 📷

| Before  | After  | After 2 
| ------- | ------- | ------- |
| ![Before-Db-Settings](https://github.com/user-attachments/assets/4233a5cf-45bf-4979-9cc9-0112b7884763) | ![After-Db-Settings-no-dialog](https://github.com/user-attachments/assets/3151ef2f-046d-42f0-b484-4f4b52990b24)| ![After-Db-Settings-Dialog](https://github.com/user-attachments/assets/d257b164-b564-42da-b38f-5701f64440b9)
| ![database-landing-before](https://github.com/user-attachments/assets/523245b2-cb53-4395-b3ff-ab41ac3fd570) | ![db-landing-after](https://github.com/user-attachments/assets/295724d7-6cdb-4559-af13-a416a2558931)

## How to test 🧪

### Prerequisites
- Managed Databases account capability
- LD flag: beta: false, enabled: true

### Verification steps
- Toggle LD will show Suspend and resume functionality
- Check DatabaseSettings to see new Suspend Button and Dialog behavior
- Check DatabaseLanding to see new suspend and resume actions
- Check DatabaseLanding to confirm that Suspend can only be used on active databases
- Check DatabaseLanding to confirm that the resume action can only be used on Suspending and Suspended database clusters
- Check DatabaseLanding to confirm that clusters with suspending, suspended, and resuming status do not have detail links under Cluster Label column

## As an Author I have considered 🤔

*Check all that apply*

- [ x] 👀 Doing a self review

---
## Commit message and pull request title format standards

> **Note**: Remove this section before opening the pull request
**Make sure your PR title and commit message on squash and merge are as shown below**

`<commit type>: [JIRA-ticket-number] - <description>`

**Commit Types:**
- `feat`: New feature for the user (not a part of the code, or ci, ...).
- `fix`: Bugfix for the user (not a fix to build something, ...).
- `change`: Modifying an existing visual UI instance. Such as a component or a feature.
- `refactor`: Restructuring existing code without changing its external behavior or visual UI. Typically to improve readability, maintainability, and performance.
- `test`: New tests or changes to existing tests. Does not change the production code.
- `upcoming`: A new feature that is in progress, not visible to users yet, and usually behind a feature flag.

**Example:** `feat: [M3-1234] - Allow user to view their login history`

---
